xdsl/backend/csl/print_csl.py

```diff
@@ -338,6 +338,13 @@ def print_block(self, body: Block):
                 ):
                     for o in outputs:
                         self.print(f"{self._var_use(o, 'var')};")
+                    # Search for all yield operations and match yield argument names to for argument names
+                    for blk in [true_region, false_region]:
+                        for op in blk.block.ops:
+                            if not isinstance(op, scf.Yield):
+                                continue
+                            for yield_arg, o in zip(op.arguments, outputs):
+                                self.variables[yield_arg] = self.variables[o]
```

Comment:
You should be able to grab the yield op by using `true_region.block.last_op` instead, this way we don't need to traverse the list at all.

Reply:
I like that very much, implementing it for both `scf.If` and `scf.For`